_assets/js/utils/backToTop.js

```diff
@@ -0,0 +1,73 @@
+import debounce from './debounce';
+
+// Set initial scroll pos:
+let yPos = window.pageYOffset;
+// Set up some flags to monitor intersection and whether we are scrolling up:
+let isIntersecting = true;
+let scrollUp = false;
+// Button to show or hide:
+const buttonId = 'crt-back-to-top-button';
+const button = document.getElementById(buttonId);
+// Class Names:
+const blockDisplay = 'mobile:display-block';
+const noDisplay = 'mobile:display-none';
+// Target element for intersection observer:
+const topElement = 'crt-page--sidenav';
+const observerTarget = document.getElementById(topElement);
+// Observer Options:
+const options = {
+  root: null,
```

Comment:
Whats this for?

Reply:
The options are for the intersection observer. They define what to observe, root:null means check for intersection with the entire viewport, root margin means don't add any margin around the target element, and threshold of 1 means make sure 100% of the target intersects with the viewport before firing the callback.